frontend/src/metabase/visualizations/lib/RowRenderer.js

```diff
@@ -159,24 +158,15 @@ export default function rowRenderer(
     chart.margins().bottom += axisLabelHeight;
   }
 
-  // cap number of rows to fit
-  const rects = chart.selectAll(".row rect")[0];
-  const containerHeight =
-    rects[rects.length - 1].getBoundingClientRect().bottom -
-    rects[0].getBoundingClientRect().top;
-  const maxTextHeight = Math.max(
-    ...chart
-      .selectAll("g.row text")[0]
-      .map(e => e.getBoundingClientRect().height),
-  );
-  const rowHeight = maxTextHeight + chart.gap() + labelPadVertical * 2;
-  const cap = Math.max(1, Math.floor(containerHeight / rowHeight));
+  const { top, bottom } = chart.margins();
+  const boundsHeight = chart.height() - top - bottom;
+  const rowHeight = (boundsHeight - chart.gap()) / group.size();
+  const barHeight = rowHeight - chart.gap();
+  const cap = Math.max(1, Math.floor(boundsHeight / ROW_MAX_HEIGHT));
   chart.cap(cap);
 
-  // assume all bars are same height?
-  const barHeight = chart.select("g.row")[0][0].getBoundingClientRect().height;
   if (barHeight > ROW_MAX_HEIGHT) {
-    const reasolableMaxGap = containerHeight / 3;
+    const reasolableMaxGap = boundsHeight / 3;
```

Comment:
i'm guessing "reasolable" is a type for `reasonable`?

Reply:
Good catch, fixed that